pulsar/consumer_partition.go

```diff
@@ -143,10 +144,61 @@ type partitionConsumer struct {
 	dlq         *dlqRouter
 
 	log log.Logger
-
+  providersMutex       sync.RWMutex
 	compressionProviders sync.Map //map[pb.CompressionType]compression.Provider
 	metrics              *internal.LeveledMetrics
 	decryptor            cryptointernal.Decryptor
+	schemaInfoCache      *schemaInfoCache
+}
+
+type schemaInfoCache struct {
+	lock   sync.RWMutex
+	cache  map[string]Schema
+	client *client
+	topic  string
+}
+
+func newSchemaInfoCache(client *client, topic string) *schemaInfoCache {
+	return &schemaInfoCache{
+		cache:  make(map[string]Schema),
+		client: client,
+		topic:  topic,
+	}
+}
+
+func (s *schemaInfoCache) Get(schemaVersion []byte) (schema Schema, err error) {
+	key := hex.EncodeToString(schemaVersion)
+	s.lock.RLock()
+	schema = s.cache[key]
+	s.lock.RUnlock()
+	if schema != nil {
+		return
+	}
+
+	pbSchema, err := s.client.lookupService.GetSchema(s.topic, schemaVersion)
+	if err != nil {
+		return nil, err
+	}
+
+	var properties = make(map[string]string)
+	if pbSchema.Properties != nil {
+		for _, entry := range pbSchema.Properties {
+			properties[*entry.Key] = properties[*entry.Value]
```

Comment:
since Key and Value are both declared as *string, do you need check the reference is not nil before assigning them to properties?

Reply:
I found a function do the same job [ConvertToStringMap](https://github.com/apache/pulsar-client-go/blob/a119bab0f8598601c0eb7f0fcd97da7ab06700c7/pulsar/internal/commands.go#/L302), but no check nil. May be I should add  nil check in ConvertToStringMap function, then invoker it.